src/Symfony/Component/HttpClient/RetryableHttpClient.php

```diff
@@ -53,41 +54,73 @@ public function __construct(HttpClientInterface $client, RetryDeciderInterface $
 
     public function request(string $method, string $url, array $options = []): ResponseInterface
     {
+        if ($this->maxRetries <= 0) {
+            return $this->client->request($method, $url, $options);
+        }
+
         $retryCount = 0;
+        $content = '';
+        $firstChunk = null;
 
-        return new AsyncResponse($this->client, $method, $url, $options, function (ChunkInterface $chunk, AsyncContext $context) use ($method, $url, $options, &$retryCount) {
+        return new AsyncResponse($this->client, $method, $url, $options, function (ChunkInterface $chunk, AsyncContext $context) use ($method, $url, $options, &$retryCount, &$content, &$firstChunk) {
             $exception = null;
             try {
                 if ($chunk->isTimeout() || null !== $chunk->getInformationalStatus()) {
                     yield $chunk;
 
                     return;
                 }
-
-                // only retry first chunk
-                if (!$chunk->isFirst()) {
-                    $context->passthru();
-                    yield $chunk;
-
-                    return;
-                }
             } catch (TransportExceptionInterface $exception) {
                 // catch TransportExceptionInterface to send it to strategy.
             }
 
             $statusCode = $context->getStatusCode();
             $headers = $context->getHeaders();
-            if ($retryCount >= $this->maxRetries || !$this->decider->shouldRetry($method, $url, $options, $partialResponse = new MockResponse($context->getContent(), ['http_code' => $statusCode, 'headers' => $headers]), $exception)) {
-                $context->passthru();
-                yield $chunk;
-
-                return;
+            if (null === $exception) {
+                if ($chunk instanceof FirstChunk) {
```

Comment:
`if ($chunk->isFirst()) {` (`FirstChunk` is an implementation detail)
same for LastChunk